### PR TITLE
fix(ui): Fix "Using Your Login". Fixes #4707

### LINKED ISSUE #4707: Fix "Using Your Login With The CLI"
## Summary 

image

Should be `ARGO_SECURE=true` and `export ARGO_SERVER=`workflows.apps.argoproj.io:443`

## Diagnostics

What Kubernetes provider are you using? 

What version of Argo Workflows are you running? 

```yaml
Paste a workflow that reproduces the bug, including status:
kubectl get wf -o yaml ${workflow} 
```

```
Paste the logs from the workflow controller:
kubectl logs -n argo $(kubectl get pods -l app=workflow-controller -n argo -o name) | grep ${workflow}
```

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

### PR DESCRIPTION
**Pull request by alexec**
Signed-off-by: Alex Collins <alex_collins@intuit.com>
Fixes #4707

Checklist:

* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo/blob/master/USERS.md).

### CODE REVIEW

File: ui/src/app/userinfo/components/cli-help.tsx [hunk 1]
```diff
@@ -2,18 +2,19 @@ import * as React from 'react';
 import {createRef, useState} from 'react';
 
 export const CliHelp = () => {
-    const argoSecure = document.location.protocol === 'https';
+    const argoSecure = document.location.protocol === 'https:';
```
Comment (alexec) [inline ui/src/app/userinfo/components/cli-help.tsx:5]: bug #1 - just did not work with https

File: ui/src/app/userinfo/components/cli-help.tsx [hunk 2]
```diff
@@ -2,18 +2,19 @@ import * as React from 'react';
...
     const argoBaseHref = document
         .getElementsByTagName('base')[0]
         .href.toString()
         .replace(document.location.protocol + '//' + document.location.host + '/', '');
-    const argoToken = decodeURIComponent(document.cookie)
-        .split(';')
-        .map(x => x.trim())
-        .find(x => x.startsWith('authorization='))
-        .replace('authorization=', '');
+    const argoToken = (
```
Comment (alexec) [inline ui/src/app/userinfo/components/cli-help.tsx:10]: bug #2 - did not work if there was no token (crashed page)

File: ui/src/app/userinfo/components/cli-help.tsx [hunk 3]
```diff
@@ -2,18 +2,19 @@ import * as React from 'react';
...
+        decodeURIComponent(document.cookie)
+            .split(';')
+            .map(x => x.trim())
+            .find(x => x.startsWith('authorization=')) || ''
+    ).replace('authorization=', '');
 
-    const text = `export ARGO_SERVER='${document.location.host}' 
+    const text = `export ARGO_SERVER='${document.location.hostname}:${document.location.port || (argoSecure ? 443 : 80)}'
```
Comment (alexec) [inline ui/src/app/userinfo/components/cli-help.tsx:17]: bug #3 - did not work if the port was the default for the protocol

**Review by jessesuen — APPROVED** [review]

**Comment by alexec** [comment]
Flakey test: TestCLIWithServerOverHTTP1Suite/TestSynchronizationWfLevelMutex  @sarabala1979

**Pull request merged**